components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -361,6 +366,39 @@ void SolanaTxManager::MakeTokenProgramTransferTxData(
           std::move(callback)));
 }
 
+void SolanaTxManager::MakeTokenProgramTxDataFromMessage(
+    const std::string& message,
+    const mojom::TransactionType tx_type,
+    mojom::SolanaSendTransactionOptionsPtr send_options,
+    MakeTokenProgramTxDataFromMessageCallback callback) {
+  absl::optional<std::vector<std::uint8_t>> message_bytes =
+      base::Base64Decode(message);
+  if (!message_bytes || message_bytes->empty() ||
+      message_bytes->size() > kSolanaMaxTxSize) {
+    std::move(callback).Run(
+        nullptr, mojom::SolanaProviderError::kInternalError,
+        l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+    return;
+  }
+
+  auto transaction =
+      SolanaTransaction::FromSignedTransactionBytes(*message_bytes);
+  transaction->set_tx_type(std::move(tx_type));
+  transaction->message()->set_recent_blockhash("");
```

Comment:
Why do we clear out their blockhash?

Reply:
I was falsely making the assumption that we should fetch the latest blockhash and use that value whenever the transaction will be approved. We don't rely on such assumptions, so I have removed it.